playwright/_impl/_helper.py

```diff
@@ -157,6 +157,10 @@ def url_matches(
             base_url = re.sub(r"^http", "ws", base_url)
         if base_url:
             match = urljoin(base_url, match)
+        parsed = urlparse(match)
+        if parsed.path == "":
+            parsed = parsed._replace(path="/")
```

Comment:
`_replace` seems to be the recommendation as per [here](https://docs.python.org/3/library/urllib.parse.html).